{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "cafa9447-8fca-4945-aa6c-d73bb2dcb64e",
   "metadata": {},
   "source": [
    "# Getting Started with Triton Notebook"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e0a0568d-651a-4fd8-8e7e-4171bfebf85c",
   "metadata": {},
   "source": [
    "A cell is either **Markdown mode** or **Code mode** as indicated in the toolbar above. Code mode is where we enter shell command or Python/R code.\n",
    "\n",
    "\n",
    "Type `%%bash` at the top of a cell to enter a shell command/script."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "804c5217-b27b-4335-a6d3-94021e3e7f92",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "hello world\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "echo \"hello world\""
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e9b11938-04bd-4dbb-9c04-3da4d3dce8ae",
   "metadata": {},
   "source": [
    "# Bioinformatics notebook"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "07a9e542-f3b2-49e2-be07-77837f98268b",
   "metadata": {},
   "source": [
    "Here is a list of bioinformatics tools available in the system."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "24398fa1-2d52-4f4a-b7e4-a198e8cdaabf",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "abba-baba\n",
      "abundance_estimates_to_matrix.pl\n",
      "ace2sam\n",
      "ajv\n",
      "align_and_estimate_abundance.pl\n",
      "alignGFA\n",
      "analyze_blastPlus_topHit_coverage.pl\n",
      "analyze_diff_expr.pl\n",
      "annotateBed\n",
      "arborist\n",
      "bakta\n",
      "bakta_db\n",
      "bamleftalign\n",
      "bamToBed\n",
      "bamToFastq\n",
      "bcftools\n",
      "bed12ToBed6\n",
      "bed2region\n",
      "bedpeToBam\n",
      "bedToBam\n",
      "bedToIgv\n",
      "bedtools\n",
      "bFst\n",
      "bgzip\n",
      "bgziptabix\n",
      "bioawk\n",
      "blast2sam.pl\n",
      "bogart\n",
      "bowtie2sam.pl\n",
      "bwa\n",
      "bwa-mem2\n",
      "bwa-mem2.avx\n",
      "bwa-mem2.avx2\n",
      "bwa-mem2.avx512bw\n",
      "bwa-mem2.sse41\n",
      "bwa-mem2.sse42\n",
      "canu\n",
      "canu.defaults\n",
      "canu-time\n",
      "cds-mapping-stats\n",
      "cds-subgraphs\n",
      "checkm\n",
      ".checkm-genome-post-link.sh\n",
      ".checkm-genome-pre-unlink.sh\n",
      "check-updates-utils\n",
      "closestBed\n",
      "clusterBed\n",
      "color-chrs.pl\n",
      "color-support\n",
      "combineMUMs\n",
      "complementBed\n",
      "contig_ExN50_statistic.pl\n",
      "convert-sam-for-rsem\n",
      "corepack\n",
      "coronaspades.py\n",
      "correctOverlaps\n",
      "coverageBed\n",
      "coverage_to_regions.py\n",
      "cuffcompare\n",
      "cuffdiff\n",
      "cufflinks\n",
      "cuffmerge\n",
      "cuffnorm\n",
      "cuffquant\n",
      "define_clusters_by_cutting_tree.pl\n",
      "delta-filter\n",
      "dnadiff\n",
      "draw-tig\n",
      "dumpBlob\n",
      "dumpContigsFromHeader\n",
      "edalign\n",
      "env_parallel\n",
      "env_parallel.ash\n",
      "env_parallel.bash\n",
      "env_parallel.csh\n",
      "env_parallel.dash\n",
      "env_parallel.fish\n",
      "env_parallel.ksh\n",
      "env_parallel.mksh\n",
      "env_parallel.pdksh\n",
      "env_parallel.sh\n",
      "env_parallel.tcsh\n",
      "env_parallel.zsh\n",
      "errorEstimate\n",
      "esparse\n",
      "esparse\n",
      "esvalidate\n",
      "esvalidate\n",
      "exact-tandems\n",
      "expandCols\n",
      "export2sam.pl\n",
      "extract_exons.py\n",
      "extract_splice_sites.py\n",
      "extract_supertranscript_from_reference.py\n",
      "extract-transcript-to-gene-map-from-trinity\n",
      "falconsense\n",
      "fastaFromBed\n",
      "fasta_generate_regions.py\n",
      "fasta-sanitize.pl\n",
      "fastp\n",
      "fastqc\n",
      "fill-aa\n",
      "fill-an-ac\n",
      "fill-fs\n",
      "fill-ref-md5\n",
      "filterCorrectionLayouts\n",
      "filterCorrectionOverlaps\n",
      "filter_low_expr_transcripts.pl\n",
      "findErrors\n",
      "findErrors-Dump\n",
      "fixErrors\n",
      "flankBed\n",
      "flye\n",
      "flye-minimap2\n",
      "flye-modules\n",
      "flye-samtools\n",
      "freebayes\n",
      "freebayes-parallel\n",
      "gatk\n",
      ".gatk-post-link.sh\n",
      "gatk-register\n",
      "generateCorrectionLayouts\n",
      "generate_freebayes_region_scripts.sh\n",
      "GenomeAnalysisTK\n",
      "genomeCoverageBed\n",
      "genotypeSummary\n",
      "getOverlap\n",
      "get_Trinity_gene_to_trans_map.pl\n",
      "gfatools\n",
      "gff2gff.py\n",
      "gffread\n",
      "gffutils-cli\n",
      "go\n",
      "gofmt\n",
      "groupBy\n",
      "gtf_to_sam\n",
      "guess-ploidy.py\n",
      "hapLrt\n",
      "hifiasm\n",
      "hisat2\n",
      "hisat2-align-l\n",
      "hisat2-align-s\n",
      "hisat2-build\n",
      "hisat2-build-l\n",
      "hisat2-build-s\n",
      "hisat2_extract_exons.py\n",
      "hisat2_extract_snps_haplotypes_UCSC.py\n",
      "hisat2_extract_snps_haplotypes_VCF.py\n",
      "hisat2_extract_splice_sites.py\n",
      "hisat2-inspect\n",
      "hisat2-inspect-l\n",
      "hisat2-inspect-s\n",
      "hisat2_read_statistics.py\n",
      "hisat2_simulate_reads.py\n",
      "htseq-count\n",
      "htseq-count-barcodes\n",
      "htseq-qa\n",
      "htsfile\n",
      "icarus.py\n",
      "iHS\n",
      "insilico_read_normalization.pl\n",
      "installed-package-contents\n",
      "interpolate_sam.pl\n",
      "intersectBed\n",
      "jop\n",
      "jParser.sh\n",
      "json5\n",
      "js-yaml\n",
      "js-yaml\n",
      "julia\n",
      "jupyter-repo2docker\n",
      "kallisto\n",
      "layoutReads\n",
      "libamd.so\n",
      "libamd.so.2\n",
      "libatomic.so\n",
      "libatomic.so.1\n",
      "libbtf.so\n",
      "libbtf.so.1\n",
      "libcamd.so\n",
      "libcamd.so.2\n",
      "libccolamd.so\n",
      "libccolamd.so.2\n",
      "libcholmod.so\n",
      "libcholmod.so.3\n",
      "libcolamd.so\n",
      "libcolamd.so.2\n",
      "libcurl.so\n",
      "libcurl.so.4\n",
      "libgfortran.so\n",
      "libgfortran.so.5\n",
      "libgit2.so\n",
      "libgit2.so.1.1\n",
      "libgmp.so\n",
      "libgmp.so.10\n",
      "libgmpxx.so\n",
      "libgmpxx.so.4\n",
      "libgomp.so\n",
      "libgomp.so.1\n",
      "libjulia-internal.so\n",
      "libjulia-internal.so.1\n",
      "libjulia.so\n",
      "libjulia.so.1\n",
      "libklu.so\n",
      "libklu.so.1\n",
      "libldl.so\n",
      "libldl.so.2\n",
      "libLLVM.so\n",
      "libmbedcrypto.so\n",
      "libmbedcrypto.so.5\n",
      "libmbedtls.so\n",
      "libmbedtls.so.13\n",
      "libmbedx509.so\n",
      "libmbedx509.so.1\n",
      "libmpfr.so\n",
      "libmpfr.so.6\n",
      "libnghttp2.so\n",
      "libnghttp2.so.14\n",
      "libopenblas64_.so\n",
      "libopenblas64_.so.0\n",
      "libopenlibm.so\n",
      "libopenlibm.so.3\n",
      "libpcre2-8.so\n",
      "libpcre2-8.so.0\n",
      "libquadmath.so\n",
      "libquadmath.so.0\n",
      "librbio.so\n",
      "librbio.so.2\n",
      "libspqr.so\n",
      "libspqr.so.2\n",
      "libssh2.so\n",
      "libssh2.so.1\n",
      "libstdc++.so\n",
      "libstdc++.so.6\n",
      "libsuitesparseconfig.so\n",
      "libsuitesparseconfig.so.5\n",
      "libumfpack.so\n",
      "libumfpack.so.5\n",
      "libunwind.so\n",
      "libunwind.so.8\n",
      "libuv.so\n",
      "libuv.so.2\n",
      "libz.so\n",
      "libz.so.1\n",
      "linksBed\n",
      "loadCorrectedReads\n",
      "loadErates\n",
      "loadTrimmedReads\n",
      "loose-envify\n",
      "mag-improve\n",
      "mapBed\n",
      "mapview\n",
      "maq2sam-long\n",
      "maq2sam-short\n",
      "markdown-it\n",
      "maskFastaFromBed\n",
      "md5fa\n",
      "md5sum-lite\n",
      "meltEHH\n",
      "mergeBed\n",
      "mergeRanges\n",
      "meryl\n",
      "metaplasmidspades.py\n",
      "metaquast\n",
      "metaquast.py\n",
      "metaspades.py\n",
      "metaviralspades.py\n",
      "mgaps\n",
      "mhapConvert\n",
      "mime\n",
      "miniasm\n",
      "minidot\n",
      "minigraph\n",
      "minimap2\n",
      "mkdirp\n",
      "mkdirp\n",
      "mmapConvert\n",
      "mmseqs\n",
      "mosdepth\n",
      "multiBamCov\n",
      "multiIntersectBed\n",
      "multiqc\n",
      "mummer\n",
      "mummerplot\n",
      "nanopolish\n",
      "nanopolish_makerange.py\n",
      "nanopolish_merge.py\n",
      "niceload\n",
      "node-gyp\n",
      "node-which\n",
      "nopt\n",
      "normalize-iHS\n",
      "novo2sam.pl\n",
      "npm\n",
      "npm-packlist\n",
      "npx\n",
      "nucBed\n",
      "nucmer\n",
      "opener\n",
      "overlapAlign\n",
      "overlapCheck\n",
      "overlapConvert\n",
      "overlapImport\n",
      "overlapInCore\n",
      "overlapInCorePartition\n",
      "overlapPair\n",
      "ovStoreBucketizer\n",
      "ovStoreBuild\n",
      "ovStoreConfig\n",
      "ovStoreDump\n",
      "ovStoreIndexer\n",
      "ovStoreSorter\n",
      "ovStoreStats\n",
      "pacote\n",
      "paf2gfa\n",
      "paftools.js\n",
      "pairToBed\n",
      "pairToPair\n",
      "parallel\n",
      "parcat\n",
      "parset\n",
      "parsort\n",
      "permuteGPAT++\n",
      "permuteSmooth\n",
      "pFst\n",
      "pilon\n",
      "plasmidspades.py\n",
      "plot-ampliconstats\n",
      "plot-bamstats\n",
      "plotHaps\n",
      "plot-roh.py\n",
      "plot-vcfstats\n",
      "pnpm\n",
      "pnpx\n",
      "popStats\n",
      "prebuild-install\n",
      "prefixEditDistance-matchLimitGenerate\n",
      "prokka\n",
      "prokka-abricate_to_fasta_db\n",
      "prokka-biocyc_to_fasta_db\n",
      "prokka-build_kingdom_dbs\n",
      "prokka-cdd_to_hmm\n",
      "prokka-clusters_to_hmm\n",
      "prokka-genbank_to_fasta_db\n",
      "prokka-genpept_to_fasta_db\n",
      "prokka-hamap_to_hmm\n",
      "prokka-tigrfams_to_hmm\n",
      "prokka-uniprot_to_fasta_db\n",
      "promer\n",
      "psl2sam.pl\n",
      "PtR\n",
      "pVst\n",
      "qrcode-terminal\n",
      "qualfa2fq.pl\n",
      "quast\n",
      "quast-download-busco\n",
      "quast-download-gridss\n",
      "quast-download-silva\n",
      "quast-lg.py\n",
      ".quast-post-link.sh\n",
      ".quast-pre-unlink.sh\n",
      "quast.py\n",
      "randomBed\n",
      "rc\n",
      "repeat-match\n",
      "repo2docker\n",
      "resolve\n",
      "retrieve_sequences_from_fasta.pl\n",
      "rimraf\n",
      "rimraf\n",
      "rimraf\n",
      "rnaspades.py\n",
      "rnaviralspades.py\n",
      "rsem-bam2readdepth\n",
      "rsem-bam2wig\n",
      "rsem-build-read-index\n",
      "rsem-calculate-credibility-intervals\n",
      "rsem-calculate-expression\n",
      "rsem-control-fdr\n",
      "rsem-extract-reference-transcripts\n",
      "rsem-for-ebseq-calculate-clustering-info\n",
      "rsem-for-ebseq-find-DE\n",
      "rsem-for-ebseq-generate-ngvector-from-clustering-info\n",
      "rsem-generate-data-matrix\n",
      "rsem-gen-transcript-plots\n",
      "rsem-get-unique\n",
      "rsem-gff3-to-gtf\n",
      "rsem-parse-alignments\n",
      "rsem-plot-model\n",
      "rsem-plot-transcript-wiggles\n",
      "rsem-prepare-reference\n",
      "rsem-preref\n",
      "rsem-refseq-extract-primary-assembly\n",
      "rsem-run-ebseq\n",
      "rsem-run-em\n",
      "rsem-run-gibbs\n",
      "rsem-sam-validator\n",
      "rsem-scan-for-paired-end-reads\n",
      "rsem-simulate-reads\n",
      "rsem-synthesis-reference-transcripts\n",
      "rsem-tbam2gbam\n",
      "run_DE_analysis_from_samples_file.pl\n",
      "run_DE_analysis.pl\n",
      "run-mummer1\n",
      "run-mummer3\n",
      "run-roh.pl\n",
      "salmon\n",
      "sam2vcf.pl\n",
      "sam_add_rg.pl\n",
      "samtools\n",
      "samtools.pl\n",
      "sanitize_unicode\n",
      "sdust\n",
      "segmentFst\n",
      "segmentIhs\n",
      "sem\n",
      "semver\n",
      "semver\n",
      "semver\n",
      "semver\n",
      "seq_cache_populate.pl\n",
      "seqkit\n",
      "seqtk\n",
      "seqtk-trinity\n",
      "sequenceDiversity\n",
      "shiftBed\n",
      "show-aligns\n",
      "show-coords\n",
      "show-diff\n",
      "show-snps\n",
      "show-tiling\n",
      "shuffleBed\n",
      "sift_bam_max_cov.pl\n",
      "slopBed\n",
      "smoother\n",
      "soap2sam.pl\n",
      "sortBed\n",
      "spades-bwa\n",
      "spades-convert-bin-to-fasta\n",
      "spades-core\n",
      "spades-corrector-core\n",
      "spades-gbuilder\n",
      "spades-gmapper\n",
      "spades-gsimplifier\n",
      "spades-hammer\n",
      "spades_init.py\n",
      "spades-ionhammer\n",
      "spades-kmercount\n",
      "spades-kmer-estimating\n",
      ".spades-post-link.sh\n",
      "spades.py\n",
      "spades-read-filter\n",
      "spades-truseq-scfcorrection\n",
      "spaligner\n",
      "splitHaplotype\n",
      "splitReads\n",
      "split_ref_by_bai_datasize.py\n",
      "sql\n",
      "sqStoreCreate\n",
      "sqStoreDumpFASTQ\n",
      "sqStoreDumpMetaData\n",
      "STAR\n",
      "STARlong\n",
      "strip-indent\n",
      "subtractBed\n",
      "svgo\n",
      "svg-term\n",
      "tabix\n",
      "tagBam\n",
      "tgStoreCompress\n",
      "tgStoreDump\n",
      "tgStoreLoad\n",
      "tgTigDisplay\n",
      "tldrl\n",
      "tldr-lint\n",
      "trimReads\n",
      "Trinity\n",
      "Trinity_gene_splice_modeler.py\n",
      "TrinityStats.pl\n",
      "truspades.py\n",
      "trycycler\n",
      "tsc\n",
      "tsserver\n",
      "unicycler\n",
      "unionBedGraphs\n",
      "update_version.sh\n",
      "utgcns\n",
      "uuid\n",
      "vcf2bed.py\n",
      "vcf2dag\n",
      "vcf2fasta\n",
      "vcf2sqlite.py\n",
      "vcf2tsv\n",
      "vcfaddinfo\n",
      "vcfafpath\n",
      "vcfallelicprimitives\n",
      "vcfaltcount\n",
      "vcfanno\n",
      "vcf-annotate\n",
      "vcfannotate\n",
      "vcfannotategenotypes\n",
      "vcfbiallelic\n",
      "vcfbreakmulti\n",
      "vcfcat\n",
      "vcfcheck\n",
      "vcfclassify\n",
      "vcfcleancomplex\n",
      "vcfclearid\n",
      "vcfclearinfo\n",
      "vcfcombine\n",
      "vcfcommonsamples\n",
      "vcf-compare\n",
      "vcfcomplex\n",
      "vcf-concat\n",
      "vcf-consensus\n",
      "vcf-contrast\n",
      "vcf-convert\n",
      "vcfcountalleles\n",
      "vcfcreatemulti\n",
      "vcfdistance\n",
      "vcfecho\n",
      "vcfentropy\n",
      "vcfevenregions\n",
      "vcffilter\n",
      "vcffirstheader\n",
      "vcf-fix-newlines\n",
      "vcf-fix-ploidy\n",
      "vcffixup\n",
      "vcfflatten\n",
      "vcfgeno2alleles\n",
      "vcfgeno2haplo\n",
      "vcfgenosamplenames\n",
      "vcfgenosummarize\n",
      "vcfgenotypecompare\n",
      "vcfgenotypes\n",
      "vcfglbound\n",
      "vcfglxgt\n",
      "vcfgtcompare.sh\n",
      "vcfhetcount\n",
      "vcfhethomratio\n",
      "vcfindelproximity\n",
      "vcfindels\n",
      "vcf-indel-stats\n",
      "vcfindex\n",
      "vcfinfo2qual\n",
      "vcfinfosummarize\n",
      "vcfintersect\n",
      "vcf-isec\n",
      "vcfjoincalls\n",
      "vcfkeepgeno\n",
      "vcfkeepinfo\n",
      "vcfkeepsamples\n",
      "vcfld\n",
      "vcfleftalign\n",
      "vcflength\n",
      "vcf-merge\n",
      "vcfmultiallelic\n",
      "vcfmultiway\n",
      "vcfmultiwayscripts\n",
      "vcfnobiallelicsnps\n",
      "vcfnoindels\n",
      "vcfnosnps\n",
      "vcfnulldotslashdot\n",
      "vcfnumalt\n",
      "vcfoverlay\n",
      "vcfparsealts\n",
      "vcf-phased-join\n",
      "vcfplotaltdiscrepancy.r\n",
      "vcfplotaltdiscrepancy.sh\n",
      "vcfplotsitediscrepancy.r\n",
      "vcfplottstv.sh\n",
      "vcfprimers\n",
      "vcfprintaltdiscrepancy.r\n",
      "vcfprintaltdiscrepancy.sh\n",
      "vcfqual2info\n",
      "vcfqualfilter\n",
      "vcf-query\n",
      "vcfrandom\n",
      "vcfrandomsample\n",
      "vcfregionreduce\n",
      "vcfregionreduce_and_cut\n",
      "vcfregionreduce_pipe\n",
      "vcfregionreduce_uncompressed\n",
      "vcfremap\n",
      "vcfremoveaberrantgenotypes\n",
      "vcfremovenonATGC\n",
      "vcfremovesamples\n",
      "vcfroc\n",
      "vcfsample2info\n",
      "vcfsamplediff\n",
      "vcfsamplenames\n",
      "vcf-shuffle-cols\n",
      "vcfsitesummarize\n",
      "vcfsnps\n",
      "vcf-sort\n",
      "vcfsort\n",
      "vcf-stats\n",
      "vcfstats\n",
      "vcfstreamsort\n",
      "vcf_strip_extra_headers\n",
      "vcf-subset\n",
      "vcftools\n",
      "vcf-to-tab\n",
      "vcf-tstv\n",
      "vcfuniq\n",
      "vcfuniqalleles\n",
      "vcfutils.pl\n",
      "vcf-validator\n",
      "vcfvarstats\n",
      "vsce\n",
      "wcFst\n",
      "wgsim\n",
      "wgsim_eval.pl\n",
      "which\n",
      "windowBed\n",
      "windowMaker\n",
      "wtdbg2\n",
      "wtdbg-cns\n",
      "wtpoa-cns\n",
      "xa2multi.pl\n",
      "yak\n",
      "yarn\n",
      "yarnpkg\n",
      "youtube-dl\n",
      "zoom2sam.pl\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "find ~/.local/bin -type f -or -type l -executable -print0 | xargs -0 basename -a | sort"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "a9982a77-7031-43da-9f01-685ba251dd69",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.2"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
